src/Traits/ManagesItemsTrait.php

```diff
@@ -393,4 +396,82 @@ protected function performProtectedCheck($item)
             throw new ModifyingProtectedValueException("Cannot access $item because it is protected");
         }
     }
+
+    /**
+     * Hydrate with external data
+     *
+     * @param $type  string    The type of data to be hydrated into the manager
+     * @param $data string     The data to be hydrated into the manager
+     * @return $this
+     * @throws \Michaels\Manager\Exceptions\SerializationTypeNotSupportedException
+     */
+    public function hydrateFrom($type, $data)
+    {
+        // we can possibly do some polymorphism for any other serialization types later
+        if($type !== 'json'){
+            throw new SerializationTypeNotSupportedException("$type serialization is not supported.");
+        }
+
+        $decodedData = $this->decodeToJson($data);
+
+        if ($this->isJson($decodedData)){
+
+            $this->reset($decodedData);
+            return $this;
+        }
+        throw new IncorrectDataException("The data is not proper JSON");
+    }
+
+    /**
+     * Hydrate with external data, appending to current data
+     *
+     * @param $type  string    The type of data to be hydrated into the manager
+     * @param $data string     The data to be hydrated into the manager
+     * @return $this
+     * @throws \Michaels\Manager\Exceptions\SerializationTypeNotSupportedException
+     *
+     */
+    public function appendFrom($type, $data)
+    {
+        if($type !== 'json'){
+            throw new SerializationTypeNotSupportedException("$type serialization is not supported.");
+        }
+
+        $decodedData = $this->decodeToJson($data);
+
+        if ($this->isJson($decodedData)){
+
+            $this->add($decodedData);
+            return $this;
+        }
+        throw new IncorrectDataException("The data is not proper JSON");
+    }
+
+    /**
+     * Checks if the input is really a json string
+     * @param $data mixed|null
+     * @return bool
+     */
+    private function isJson($data)
```

Comment:
I don't know about calling this `isJson()` because that's not what its really asking. The data should be an array at this point, not json. Maybe something like `validateDecodedData()` or `noJsonErrors()`